src/EventEndingRow.js

```diff
@@ -75,10 +75,28 @@ class EventEndingRow extends React.Component {
   }
 
   renderShowMore(segments, slot) {
-    let { localizer, slotMetrics } = this.props
+    let { localizer, slotMetrics, components } = this.props
     const events = slotMetrics.getEventsForSlot(slot)
     const remainingEvents = eventsInSlot(segments, slot)
     const count = remainingEvents.length
+
+    if (components && components.showMore) {
+      const ShowMore = components.showMore
+
+      return count ? (
+        <ShowMore
+          localizer={localizer}
+          slotDate={slotMetrics.getDateForSlot(slot - 1)}
```

Comment:
The `getDateForSlot` access the slots range, and array, using the slot as the index. But the slot received in this method are not zero-based, which means, we need to adjust for the expected zero-based index used to pull from the rage: https://github.com/davidmh/react-big-calendar/blob/770be8b87118a59f7be01f0ea44e9bead14b59fa/src/utils/DateSlotMetrics.js#L38-L40

Reply:
@davidmh Looks like a good catch to me. This code was long before my time on the project, and I can see where it would have been easily missed.

Reply:
I started looking into the index issue with the slots, but that ends up breaking a bunch of other features, so I'll just leave it as it is.

I added a few more events in the story to make sure the custom component behaves as expected on the first day of a row.